### DISCUSSION TITLE
[New Feature] Ballerina formatter support for optimising imports

### DISCUSSION THREAD

**Discussion by AzeemMuzammil** [ballerina-platform/ballerina-lang#39307] [Ideas]
Optimizing the imports in a source file gives much readability and a common formatting feature supported by many IDEs for almost all languages. And so far Ballerina formatter does not have import-optimization feature. And we have created a document to compare how import optimization works with different languages so we can come up with a strategy for Ballerina.

Here I have attached the link to the document anyone can refer to: https://docs.google.com/document/d/1eu3V6RQUiBRNCloT5TzSHeG63vxzWPNB5xBxQb2uiMU/edit?usp=sharing

@hevayo @sameerajayasoma @mohanvive you inputs in this regard is much appreciated.

**Comment by malinthar** [comment]
Related issue https://github.com/ballerina-platform/ballerina-lang/issues/38296
